docs/docs/ports.md

```diff
@@ -0,0 +1,234 @@
+
+## Random Port Assignment
+
+Using the value 0 for any port settings indicates to Marathon that you would like a random port assignment. However, if containerPort is set to 0 within a portMapping, it is set to the same value as hostPort.
+
+## Environment Variables
+
+Each port value is exposed to the running application instance via environment variables `$PORT0`, `$PORT1`, etc. Each Marathon application is given a single port by default, so `$PORT0` is always available. These variables are available inside a Docker container being run by Marathon too.
+
+## Example Configuration
+
+### Host Mode
+
+Host mode networking is the default networking mode for Docker containers and the only networking mode for non-Docker applications. Note that it not necessary to `EXPOSE` ports in your Dockerfile.
+
+#### Enabling Host Mode
+
+Host mode is enabled by default for containers. If you wish to be explicit, you can also specify it manually through the `network` property:
+
+```json
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "my-image:1.0",
+      "network": "HOST"
+    }
+  },
+```
+
+For non-Docker applications, you don't need to specify anything.
+
+#### Specifying Ports
+
+You can specify the ports that are available through the `ports` array:
+
+```json
+    "ports": [
+        0, 0, 0
+    ],
+```
+
+In this example, we specify three randomly assigned ports which would then be available to our command via the environment variables `$PORT0`, `$PORT1` and `$PORT2`.
+
+These ports also become service ports for your application and are reserved cluster wide.
+
+You can also specify specific ports:
+
+```json
+    "ports": [
+        2001, 2002, 3000
+    ],
+```
+
+In this case, `$PORT0`, `$PORT1` and `$PORT3` remain randomly assigned. However, the service ports for this application are now `2001`, `2002` and `3000`.
+
+If you want the actual ports available to your application, you can set `requirePorts` to `true` (`requirePorts` is `false` by default). This will tell Marathon to only schedule this application on agents which have these ports available:
+
+```json
+    "ports": [
+        2001, 2002, 3000
+    ],
+    "requirePorts" : true
+```
+
+The service ports and the environment variables`$PORT0`, `$PORT1`, and `$PORT2` are both now `2001`, `2002` and `3000` respectively.
+
+This property is useful if you don't use a service discovery solution such as HAProxy to proxy requests on service ports.
+
+#### Referencing Ports
+
+You can reference ports in the Dockerfile for our fictitious app as follows:
+
+```sh
+CMD ./my-app --http-port=$PORT0 --https-port=$PORT1 --monitoring-port=$PORT2
```

Comment:
I know that this is in the `HOST` networking section but I would mention that this only works for `HOST` networking here again. This is a constant source of confusion.